test/fake_code_signer.rb

```diff
@@ -0,0 +1,59 @@
+require File.join File.dirname(__FILE__), "fake_code_signer/pe_wrapper"
+
+# This class digitally signs an executable.
+# In reality, it doesn't create a "valid" digital signature - but it sets up the executable
+# headers correctly with the appropriate size and offsets. It also appends the "signature" to
+# the end of the executable, complete with 'padding' between the end of the file and sig.
+# Although this signature is not 'valid' in the sense that it meets the authenticode standard
+# it still meets the expectations of Ocra which only needs the appended "signature" to match
+# the size and location as specificed in the header.
+#
+# The purpose of this class is to create a "signed" executable so that we can test Ocra's
+# ability to build executables that are resilient to authenticode code signing, i.e that
+# Ocra executables still work after they've been digitally signed.
+#
+# For more information see:
+# The PE Format deep dive: https://msdn.microsoft.com/en-us/library/ms809762.aspx
+# Reverse engineering the authenticode format: https://www.cs.auckland.ac.nz/~pgut001/pubs/authenticode.txt
+class FakeCodeSigner
+  FAKE_SIG = "fake signature"
+
+  def initialize(input:, output:, padding: 4)
```

Comment:
Maybe `input_filename`, `output_filename`, and `padding_byte_count` or something? It's obvious what these arguments are in the test that uses it, but not so much here.